integration-tests/src/nft_marketplace.rs

```diff
@@ -135,3 +138,102 @@ fn create_class_should_fail_when_relay_chain_location_not_registered() {
 		);
 	});
 }
+
+#[test_case(ClassType::Marketplace, true ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, false ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, false ; "redeemable class")]
+#[test_case(ClassType::Auction, false ; "auction class")]
+#[test_case(ClassType::HydraHeads, false ; "hydra heads class")]
+fn nft_permissions_should_allow_marketplace_class_creation(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_create(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace, true ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, false ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, false ; "redeemable class")]
+#[test_case(ClassType::Auction, false ; "auction class")]
+#[test_case(ClassType::HydraHeads, false ; "hydra heads class")]
+fn nft_permissions_should_allow_marketplace_class_minting(class_type: ClassType, is_allowed: bool) {
```

Comment:
The test name does not reveal exactly what we are testing. We are testing more than that.

I am of the opinion that it is one of the exceptional situations where we can easily ditch the `ShouldWhen` template, and just writing:

`test_nft_permission_for_class_minting`

Reply:
Ok. I had a hard time trying to come up with a reasonably good test names :)

Reply:
yes, had the same, it is indeed challenging to apply ShouldWhen here. 

This is one of the situations where we should not be dogmatic, and less is more :)